### ISSUE TITLE
[Feature Request]:Add MAX support for sapientinc/HRM-Text-1B

### ISSUE THREAD

**Issue by Milan933-coder** [modular/mojo#6675]
### What is your request?

I would like MAX to add native pipeline support for
`sapientinc/HRM-Text-1B`.

This model uses the Hugging Face architecture name `HrmTextForCausalLM` and the
model type `hrm_text`. The new architecture can be added under:

```text
max/python/max/pipelines/architectures/hrm_text/
```

The architecture should follow the layout described in the MAX custom model
architecture guide:

```text
hrm_text/
|-- __init__.py
|-- arch.py
|-- model.py
|-- model_config.py
`-- weight_adapters.py
```

After support is added, users should be able to serve the model with:

```bash
max serve --model-path sapientinc/HRM-Text-1B
```

### What is your motivation for this change?

`sapientinc/HRM-Text-1B` is a small text generation model focused on reasoning.
It is useful because it gives strong reasoning results while staying much
smaller than many common LLMs.

Public model details:

- [ ] Model name: `sapientinc/HRM-Text-1B`
- [ ] Architecture: `HrmTextForCausalLM`
- [ ] Model type: `hrm_text`
- [ ] Parameters: about 1.15B
- [ ] Training data: about 40B tokens
- [ ] Hugging Face checkpoint size: about 2.37 GB
- [ ] License: Apache 2.0
- [ ] Task: text generation
- [ ] Weight format: SafeTensors

Reported 1B benchmark results from the HRM-Text project:

| Task | Score |
|------|-------|
| GSM8K | 84.7% |
| MATH | 56.5% |
| DROP | 82.3% |
| MMLU | 60.7% |
| ARC-C | 81.9% |
| HellaSwag | 63.4% |
| Winogrande | 72.4% |
| BoolQ | 86.2% |

Adding this model would give MAX users a compact reasoning model for testing
serving, accuracy, and throughput.

### Any other details?

This request is for inference support only. It is not asking for training
support or a new modality.

Suggested implementation checklist:

- [ ] Add a new `hrm_text` architecture directory.
- [ ] Add `arch.py` with a `SupportedArchitecture` entry for
      `HrmTextForCausalLM`.
- [ ] Add `model_config.py` to parse the Hugging Face config.
- [ ] Add `model.py` to build the MAX graph for HRM-Text.
- [ ] Add `weight_adapters.py` for SafeTensors weight loading.
- [ ] Register the architecture in
      `max/python/max/pipelines/architectures/__init__.py`.
- [ ] Confirm the model can run through `max serve`.

Important config values to support:

- [ ] `vocab_size: 65536`
- [ ] `hidden_size: 1536`
- [ ] `intermediate_size: 4096`
- [ ] `num_hidden_layers: 16`
- [ ] `num_attention_heads: 12`
- [ ] `num_key_value_heads: 12`
- [ ] `head_dim: 128`
- [ ] `H_cycles: 2`
- [ ] `L_cycles: 3`
- [ ] `max_position_embeddings: 4096`
- [ ] `rope_theta: 10000.0`
- [ ] `rms_norm_eps: 1e-6`
- [ ] `prefix_lm: true`
- [ ] `pad_token_id: 5`
- [ ] `bos_token_id: 6`
- [ ] `eos_token_id: 11`

Model behavior to handle:

- [ ] Support the HRM high-level and low-level Transformer loop.
- [ ] Support the repeated compute pattern using
      `H_cycles * (L_cycles + 1)`.
- [ ] Support state injection between the high-level and low-level modules.
- [ ] Support PrefixLM-style text generation behavior if required by the model.

Suggested validation checklist:

- [ ] Start development with the `--custom-architectures` flag.
- [ ] Compare MAX logits against a Transformers or PyTorch reference.
- [ ] Run cosine distance, KL divergence, and tolerance checks.
- [ ] Run `lm-eval` on `gsm8k_cot_llama` with 320 samples, seed 42, and
      temperature 0.
- [ ] Compare the MAX result with a reference backend such as vLLM or SGLang.
- [ ] Follow the MAX guide target of at least 95% of the reference score.
- [ ] If the reference GSM8K score is 84.7%, the target score should be at
      least 80.5%.
- [ ] Report basic serving numbers such as tokens per second, peak memory use,
      GPU type, dtype, and batch size.

Useful links:

- Hugging Face model: https://huggingface.co/sapientinc/HRM-Text-1B
- Hugging Face config: https://huggingface.co/sapientinc/HRM-Text-1B/blob/main/config.json
- HRM-Text project: https://github.com/sapientinc/HRM-Text
- MAX model contribution guide: /max/docs/contributing-models.md

**Comment by Milan933-coder** [comment]
Can someone please assighn it To me .

**Comment by Milan933-coder** [comment]
@tjk213 please have a look on it

**Comment by Milan933-coder** [comment]
@BradLarson  are you going to assighn this to me
